charon/src/bin/charon-driver/translate/translate_bodies.rs

```diff
@@ -566,28 +566,42 @@ impl BodyTransCtx<'_, '_, '_> {
                                     self.translate_constant_expr_to_const_generic(span, len)?;
                                 UnsizingMetadata::Length(len)
                             }
-                            hax::UnsizingMetadata::VTablePtr(impl_expr) => {
+                            hax::UnsizingMetadata::DirectVTable(impl_expr) => {
                                 let tref = self.translate_trait_impl_expr(span, impl_expr)?;
-                                match &impl_expr.r#impl {
+                                let vtable = match &impl_expr.r#impl {
                                     hax::ImplExprAtom::Concrete(tref) => {
                                         // Ensure the vtable type is translated.
-                                        let _: GlobalDeclId = self.register_item(
+                                        let id = self.register_item(
                                             span,
                                             tref,
                                             TransItemSourceKind::VTableInstance(
                                                 TraitImplSource::Normal,
                                             ),
                                         );
+                                        Some(GlobalDeclRef {
+                                            // FIXME: this is wrong, I believe
+                                            generics: Box::new(GenericArgs::empty()),
+                                            id,
+                                        })
```

Comment:
You should be able to get the right generics using `translate_item` instead of `register_item`.